Tests/FileIO/TestXmlGmlReader.cpp

```diff
@@ -19,120 +19,242 @@
+			ASSERT_EQ(name, line_name);
+		};
+
+		checkPolylineProperty(line_vec, 0, {0, 1, 2}, "left");
+		checkPolylineProperty(line_vec, 1, {3, 4, 5}, "center");
+		checkPolylineProperty(line_vec, 2, {0, 3}, "");
+		checkPolylineProperty(line_vec, 3, {3, 6}, "");
+		checkPolylineProperty(line_vec, 4, {6, 7, 8}, "right");
 	}
 
-	{   // Create surfaces.
+	void createSurfaces()
+	{
+		auto const points = geo_objects.getPointVec(geo_name);
+		const std::vector<std::size_t> pnt_id_map(
+			geo_objects.getPointVecObj(geo_name)->getIDMap());
+
 		auto sfcs = std::unique_ptr<std::vector<GeoLib::Surface*>>(
-		    new std::vector<GeoLib::Surface*>(2));
+			new std::vector<GeoLib::Surface*>(2));
+		std::map<std::string, std::size_t>* sfc_names =
+			new std::map<std::string, std::size_t>;
 		(*sfcs)[0] = new GeoLib::Surface(*points);
+		(*sfcs)[0]->addTriangle(pnt_id_map[0], pnt_id_map[3], pnt_id_map[1]);
+		(*sfcs)[0]->addTriangle(pnt_id_map[1], pnt_id_map[3], pnt_id_map[4]);
 		(*sfcs)[0]->addTriangle(pnt_id_map[1], pnt_id_map[4], pnt_id_map[2]);
 		(*sfcs)[0]->addTriangle(pnt_id_map[2], pnt_id_map[4], pnt_id_map[5]);
-		(*sfcs)[0]->addTriangle(pnt_id_map[2], pnt_id_map[5], pnt_id_map[3]);
-		(*sfcs)[0]->addTriangle(pnt_id_map[3], pnt_id_map[5], pnt_id_map[6]);
 		(*sfcs)[1] = new GeoLib::Surface(*points);
-		(*sfcs)[1]->addTriangle(pnt_id_map[4], pnt_id_map[7], pnt_id_map[9]);
-		(*sfcs)[1]->addTriangle(pnt_id_map[4], pnt_id_map[9], pnt_id_map[6]);
-		geo_objects.addSurfaceVec(std::move(sfcs), geo_name);
+		(*sfcs)[1]->addTriangle(pnt_id_map[3], pnt_id_map[6], pnt_id_map[8]);
+		(*sfcs)[1]->addTriangle(pnt_id_map[3], pnt_id_map[8], pnt_id_map[5]);
+		(*sfc_names)["SecondSurface"] = 1;
+		geo_objects.addSurfaceVec(std::move(sfcs), geo_name, sfc_names);
 	}
 
+	void checkSurfaceProperties()
+	{
+		auto checkTriangleIDs = [](
+			GeoLib::Triangle const& tri, std::array<std::size_t,3> ids)
+		{
+			ASSERT_EQ(ids[0], tri[0]);
+			ASSERT_EQ(ids[1], tri[1]);
+			ASSERT_EQ(ids[2], tri[2]);
+		};
+
+		auto checkSurface = [&checkTriangleIDs](GeoLib::SurfaceVec const& sfcs,
+			std::size_t sfc_id, std::vector<std::array<std::size_t,3>> tri_ids,
+			std::string const& name)
+		{
+			auto const& sfc_vec = *(sfcs.getVector());
+			auto const& sfc = *(sfc_vec[sfc_id]);
+			ASSERT_EQ(tri_ids.size(), sfc.getNTriangles());
+			for (std::size_t k(0); k<tri_ids.size(); ++k)
+				checkTriangleIDs(*(sfc[k]), tri_ids[k]);
+
+			std::string sfc_name;
+			sfcs.getNameOfElementByID(sfc_id, sfc_name);
+			ASSERT_EQ(0u, name.compare(sfc_name));
+		};
+
+		auto const read_sfcs = geo_objects.getSurfaceVecObj(geo_name);
+		ASSERT_EQ(2u, read_sfcs->size());
+		checkSurface(*read_sfcs, 0,
+			{{{0,3,1}}, {{1,3,4}}, {{1,4,2}}, {{2,4,5}}}, "");
+		checkSurface(*read_sfcs, 1, {{{3,6,8}}, {{3,8,5}}}, "SecondSurface");
+	}
+};
+
+TEST_F(FileIOXmlGml, QtXmlGmlWriterReaderTest)
+{
+	// Writer test
+	std::string test_data_file(BaseLib::BuildInfo::source_path + "/Tests/FileIO/xmlgmltestdata.gml");
```

Comment:
Will the test write to the source tree? I think it would be better to write to some temporary directory.